### PR TITLE
[ACR] Implement Cleopatra, Exiled Pharaoh

### PR DESCRIPTION
**Pull request by jimga150**
#11854

### CODE REVIEW

File: Mage.Sets/src/mage/cards/c/CleopatraExiledPharaoh.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.NOBLE);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(4);
+
+        // Allies -- At the beginning of your end step, put a +1/+1 counter on each of up to two other target legendary creatures.
+        // Based on Angelic Quartermaster
+        Ability ability = new BeginningOfEndStepTriggeredAbility(new AddCountersTargetEffect(CounterType.P1P1.createInstance()), TargetController.YOU, false);
+        ability.setAbilityWord(AbilityWord.ALLIES);
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/c/CleopatraExiledPharaoh.java:57]: Are you sure these are ability words? Pretty sure they should just be flavor words

File: Mage.Sets/src/mage/cards/c/CleopatraExiledPharaoh.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        ability.addTarget(new TargetPermanent(0, 2, filter));
+        this.addAbility(ability);
+
+        // Betrayal -- Whenever a legendary creature with counters on it dies, draw a card for each counter on it. You lose 2 life.
+        Ability ability2 = new DiesCreatureTriggeredAbility(new CleopatraExiledPharaohEffect(), false, filter2);
+        ability2.setAbilityWord(AbilityWord.BETRAYAL);
+        Effect effect = new LoseLifePermanentControllerEffect(2);
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/c/CleopatraExiledPharaoh.java:64]: this should just be `LoseLifeSourceControllerEffect` as it is not tied to the permanent's controller

File: Mage.Sets/src/mage/cards/c/CleopatraExiledPharaoh.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    @Override
+    public CleopatraExiledPharaoh copy() {
+        return new CleopatraExiledPharaoh(this);
+    }
+}
+
+// Based on DrizztDoUrdenEffect
+class CleopatraExiledPharaohEffect extends OneShotEffect {
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/c/CleopatraExiledPharaoh.java:81]: Use common DrawCardSourceControllerEffect with dynamic value - existing `CountersSourceCount` (`null` constructor param for any type of counter, though missing explicit doc comments on that usage).  Maybe other cards can be simplified too (as separate PR).

**Review by xenohedron — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/c/CleopatraExiledPharaoh.java (hunk 3 above)
Comment (jimga150) [reply Mage.Sets/src/mage/cards/c/CleopatraExiledPharaoh.java:81]: I dont think that can be used to get the amount of counters on the triggering permanent--would it be better to implement a new `DynamicValue` to count the counters on a triggering permanent?

Comment (xenohedron) [reply Mage.Sets/src/mage/cards/c/CleopatraExiledPharaoh.java:81]: My mistake. You are correct, need a separate dynamic value (if possible - if doesn't work for whatever reason, fine as is)

**Review by xenohedron — APPROVED** [review]

**Pull request merged**